### PR TITLE
31 pass at perm collection

### PR DESCRIPTION
**Pull request by sepowitz**

### CODE REVIEW

File: app/reducers/gallery.js [hunk 1]
```diff
@@ -24,12 +29,42 @@ export default function gallery(state=initialState, action) {
...
+      let orderedPermanents = permanents.sort(function (a, b) {
+        if (a.bumpCount > b.bumpCount) {
+          return -1;
+        }
+        if (a.bumpCount < b.bumpCount) {
+          return 1;
+        }
+        return 0;
+      });
+      console.log(orderedPermanents)
```
Comment (dhunninghake) [inline app/reducers/gallery.js]: dis can be nixed

**Comment by dhunninghake** [comment]
just a small note about console log. 
otherwise looks good! ⚡️

File: app/components/Gallery.js [hunk 2]
```diff
@@ -1,15 +1,61 @@
...
+
+  didClickImageWrapper = (media) => {
+    //Firebase success/error callback
+    let onComplete = function(error) {
+      if (error) {
+        console.log('Synchronization failed');
+      } else {
+        console.log('Synchronization succeeded');
+      }
+    };
```
Comment (hhff) [inline app/components/Gallery.js]: Do we need this onComplete function anymore?

File: app/components/Gallery.js [hunk 3]
```diff
@@ -1,15 +1,61 @@
...
+    // Check if media id is in permanents
+    let permanentsRef = this.props.firebaseRef.child('/permanents');
+
+    let match = this.props.collection.find(permanentMedia => permanentMedia.media.id === media.id);
+
+    if(match) {
+      let matchRef = this.props.firebaseRef.child('/permanents/' + match.id + '/bumpCount')
+
+      matchRef.transaction(function (currentBumpCount) {
+        return (currentBumpCount) + 1;
```
Comment (hhff) [inline app/components/Gallery.js]: Let's es6-ify this bad boy

File: app/components/PermanentGallery.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        console.log('Synchronization succeeded');
+      }
+    };
+
+    let imageRef = this.props.firebaseRef.child('/permanents/' + media.id + '/bumpCount')
+
+    imageRef.transaction(function (currentBumpCount) {
+      return (currentBumpCount) + 1;
+    }, onComplete);
+  }
```
Comment (hhff) [inline app/components/PermanentGallery.js:28]: Sames here

File: app/handlers/PermanentCollection.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+import PermanentGallery from '../components/PermanentGallery';
+
+export default class PermanentCollection extends Component {
+
+  render() {
+    return (
+      <PermanentGallery/>
+    )
+  }
+}
```
Comment (hhff) [inline app/handlers/PermanentCollection.js]: This seems redundant - we can just make `PermanentGallery` the handler right?

File: app/reducers/gallery.js (hunk 1 above)
Comment (hhff) [inline app/reducers/gallery.js]: I'd prolly write the sort like this:

```
let orderedPermanents = permanents.sort((a, b) => {
  if (a.bumpCount > b.bumpCount) { return -1; }
  if (a.bumpCount < b.bumpCount) { return 1; }
  return 0;
});
```

mainly just wanna b sure we es6-ify everything!  online-liner stuff is optional i dont mind

File: app/handlers/PermanentCollection.js (hunk 5 above)
Comment (sepowitz) [inline app/handlers/PermanentCollection.js]: not sure...

**Pull request merged**